src/account.ts

```diff
@@ -255,106 +255,97 @@ export default class EthereumAccount {
    * - Keeping the settlement methods "pure" with respect to the balance of the account greatly
    *   simplifies the balance logic (but does require committing the balance first, then refunding)
    */
   async attemptSettle (): Promise<void> {
     return this.queue.synchronize(TaskPriority.Outgoing, async () => {
-      let settlementBudget = new BigNumber(0)
-      let amountLeftover = new BigNumber(0)
-
-      // Don't attempt settlement if there's no configured settle threshold ("receive only" mode)
       const settleThreshold = this.master._balance.settleThreshold
-      if (!settleThreshold) {
-        return
-      }
 
-      const shouldSettle = settleThreshold.gt(this.account.balance)
+      // Don't settle in "receive only" mode (no settle threshold configured)
+      const shouldSettle = settleThreshold
+        && settleThreshold.gt(this.account.balance)
+        && this.account.payoutAmount.gt(0)
+
       if (!shouldSettle) {
         return
       }
 
-      settlementBudget = this.master._balance.settleTo.minus(this.account.balance)
-      if (settlementBudget.lte(0)) {
-        // This *should* never happen, since the master constructor verifies that settleTo >= settleThreshold
-        return this.master._log.error(`Critical settlement error: settlement threshold triggered, but settle amount of ` +
-          `${format(settlementBudget, Unit.Gwei)} is 0 or negative`)
+      // Determine the amount to settle for
+      const settlementBudget = BigNumber.min(
+        this.master._balance.settleTo.minus(this.account.balance),
+        // If we're not prefunding, the amount should be limited by the total packets we've fulfilled
+        // If we're prefunding, the payoutAmount is infinity, so it doesn't affect the amount to settle
+        this.account.payoutAmount
+      )
+
+      // This should never error, since settleTo < maximum
+      this.addBalance(settlementBudget)
+      this.account.payoutAmount = this.account.payoutAmount.minus(settlementBudget)
+
+      // Convert gwei -> wei
+      const settlementBudgetWei = convert(settlementBudget, Unit.Gwei, Unit.Wei).dp(0, BigNumber.ROUND_FLOOR)
+
+      const budget = format(settlementBudgetWei, Unit.Wei)
+      const account = this.account.accountName
+      this.master._log.info(`Settlement triggered with ${account} for maximum of ${budget}`)
+
+      // If the the entire budget is spent, skip the subsequent settlement function
+      const checkBudget = (f: (budget: BigNumber) => Promise<BigNumber>) =>
+        (budget: BigNumber) =>
+          budget.lte(0) ? budget : f(budget)
+
+      const settle = (budget: BigNumber) =>
+        // 1) Try to send a claim: spend the channel down entirely before funding it
+        this.sendClaim(budget)
+        // 2) Open or deposit to a channel if it's necessary to send anything leftover
+        .then(checkBudget(l => this.fundOutgoingChannel(l)))
```

Comment:
What is `l`? In general I'd prefer to avoid single-letter variable names (especially `l` because it looks so much like a `1`)